src/targets/gpu/kernels/include/migraphx/kernels/layernorm.hpp

```diff
@@ -50,10 +50,16 @@ __device__ void generic_binary_layernorm(
     reduce::block::run<reduce_output>([&](auto, auto r) {
         using value_type         = typename Input1::type;
         constexpr auto relements = r.template elements<Input1>();
+        auto relements_r         = vec_type<value_type>{1.0 / relements};
+        auto relements_rsqrt     = sqrt(relements_r);
```

Comment:
is there a way i can check if its const-folded?

Reply:
also, doing `vec_type<value_type>{rsqrt(relements)}` results in `error: non-constant-expression cannot be narrowed from type 'decltype(::rsqrt(xs))' (aka 'double') to 'vec_type<value_type>' (aka '_Float16') in initializer list [-Wc++11-narrowing]`

Reply:
Ah thats not a vector type, so `sqrt(relements_r)` should be fine. Should probably declare it as `const` at least.

Reply:
> is there a way i can check if its const-folded?

You can do `MIGRAPHX_GPU_DUMP_ASM=1 MIGRAPHX_GPU_DEBUG_SYM=1` to see if its in the assembly. With debug symbols it should be annotated with locations in the source code, although with optimizations it might be harder to track down.